### PR TITLE
Support schema version 1.2.0

### LINKED ISSUE #284: Support schema version 1.2.0
Some new fields were added:
- schema_version
- severity
- credits
- top level database_specific

### PR DESCRIPTION
**Pull request by oliverchang**
Also delete legacy code dealing with pre-1.0 schema.

Fixes #284.

### CODE REVIEW

File: lib/osv/vulnerability.proto [hunk 1]
```diff
@@ -148,14 +134,23 @@ message Affected {
...
+message Severity {
+  // Type of the severity.
+  enum Type {
+    UNSPECIFIED = 0;
+    CVSS_V3 = 1;
+  }
+  // The type of this severity entry.
+  Type type = 1;
+  // The quantitative score.
+  string score = 2;
```
Comment (inferno-chromium) [inline lib/osv/vulnerability.proto:146]: if it is quantitative score, then shouldn't this be int or float ?

File: lib/osv/vulnerability.proto [hunk 2]
```diff
@@ -148,14 +134,23 @@ message Affected {
...
+}
+
+message Credit {
+  // The name to give credit to.
+  string name = 1;
+  // Contact methods.
```
Comment (inferno-chromium) [inline lib/osv/vulnerability.proto:152]: should the comment have example of methods ?

**Review by inferno-chromium — APPROVED** [review]

File: lib/osv/vulnerability.proto (hunk 1 above)
Comment (oliverchang) [reply lib/osv/vulnerability.proto:146]: This is a string for future extensibility, or if we relax the "quantitative" requirement.

File: lib/osv/vulnerability.proto (hunk 2 above)
Comment (oliverchang) [reply lib/osv/vulnerability.proto:152]: Clarified this to be URLs. This is all better documented at https://ossf.github.io/osv-schema/ :)

**Pull request merged**